ecs.tf

```diff
@@ -40,3 +52,83 @@ resource "aws_ecs_task_definition" "waggledance" {
   container_definitions    = data.template_file.waggledance.rendered
   tags                     = var.tags
 }
+
+resource "aws_appautoscaling_target" "waggledance" {
+  count = var.wd_instance_type == "ecs" && var.enable_autoscaling ? 1 : 0
+
+  max_capacity       = var.wd_ecs_max_task_count
+  min_capacity       = var.wd_ecs_task_count
+  resource_id        = "service/${local.instance_alias}/${local.instance_alias}-service"
+  scalable_dimension = "ecs:service:DesiredCount"
+  service_namespace  = "ecs"
+
+  depends_on = [
+    aws_ecs_service.waggledance_service,
+  ]
+}
+
+resource "aws_appautoscaling_policy" "waggledance" {
+  count = var.wd_instance_type == "ecs" && var.enable_autoscaling ? 1 : 0
+
+  name               = "cpu"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_appautoscaling_target.waggledance[0].resource_id
+  scalable_dimension = aws_appautoscaling_target.waggledance[0].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.waggledance[0].service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageCPUUtilization"
+    }
+
+    target_value       = var.wd_target_cpu_percentage
+    scale_in_cooldown  = var.cpu_scale_in_cooldown
+    scale_out_cooldown = var.cpu_scale_out_cooldown
+  }
+
+  depends_on = [aws_appautoscaling_target.waggledance]
+}
+
+resource "aws_lb" "waggledance" {
+  count              = var.wd_instance_type == "ecs" && var.enable_autoscaling ? 1 : 0
+  name               = local.instance_alias
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = var.subnets
+
+  tags = var.tags
+}
+
+resource "aws_lb_target_group" "waggledance" {
+  count       = var.wd_instance_type == "ecs" && var.enable_autoscaling ? 1 : 0
+  port        = local.wd_port
+  protocol    = "TCP"
+  target_type = "ip"
+  vpc_id      = var.vpc_id
+  tags        = var.tags
+  health_check {
+    protocol = "HTTP"
+    port     = 18000
+    path     = "/actuator/health"
+    //interval = 10
+    //healthy_threshold   = 3
+    //unhealthy_threshold = 3
+    //matcher             = var.healthcheck_matcher
```

Comment:
Are these comments here because you want to enable this at some point?

Reply:
yes, cleaned up anyway..